#!/usr/bin/env python
PACKAGE = 'r2_behavior'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
# Tracking parameters
main_state_enum = gen.enum([
  gen.const("Idle",str_t,'idle',"IDLE: stay Idle"),
  gen.const("Interacting",str_t,'interacting',"INTERACTING: Interactive mode"),
  gen.const("Performing",str_t,'performing',"PERFORMING: Performing")
], "initial state")

gen.add("init_state", str_t, 0, "Initial state on startup", 'interacting', edit_method=main_state_enum)


gen.add("thinking_operator", bool_t, 0, "Think while answer is generated", True)
gen.add("thinking_time", double_t, 0, "Thinking timeout (go back to listening, if no response)", 5, 1, 10)

exit(gen.generate(PACKAGE, "states", "States"))